runtime/compiler/env/VMJ9.cpp

```diff
@@ -4086,6 +4090,30 @@ TR_J9VMBase::canDereferenceAtCompileTime(TR::SymbolReference *fieldRef, TR::Comp
       return false;
    }
 
+bool
+TR_J9VMBase::isStable(TR::Symbol *field, int cpIndex, TR_ResolvedMethod *owningMethod, TR::Compilation *comp)
+   {
+   if (cpIndex < 0)
+      return false;
+   
+   J9Class *fieldClass = (J9Class*)owningMethod->classOfMethod();
+   if (!fieldClass)
```

Comment:
Do we also need to check if the class has been initialized ?

Reply:
I think we don't have because of https://github.com/eclipse-openj9/openj9/pull/12893#discussion_r650968112

Reply:
Just to clarify a bit more... Say we are compiling `A.foo()` that references `B.f`. In this case, the `fieldClass` (might not be a good name) will be pointing to `A` and it would be loaded and initialized, I believe.  We pass `A` to the helper. Question to @tajila : what happens if `B` is not loaded?

Reply:
I suspect `B.f` would appear unresolved in our IL trees if `B` was not loaded and presumably you won't try to do any transformation in such a case. I am less clear about the scenario where `B` is loaded but not yet initialized.

Reply:
I will let @tobi respond but if the helper returns true **iff** the field is annotated as stable, I think the result will be correct. This is because we are only interested in the current value of the field and not how it was set. If the value is equal to the default value, we do not fold.

Reply:
After talking to @tajila and @jdmpapin I think I can summarize the issues:

1. Class of B not being loaded: it will be loaded since B will be a known object
2. Class of B or B itself not being initialized: definition of Stable is not based on initialization but rather on the current value of the field
3. B.f not being resolved. We currently have two passes in the code that call `isStable()`. One from `canDereferenceAtCompileTime()` which immediately checks that `B.f` is resolved. Another one is from `InterpreterEmulator`. We proved that this code ensures that the field is resolved. 

Notice that `isStable()` takes `TR::Symbol` as a parameter and not `TR::SymbolReference` so it's not easy to check whether the field was resolved or not. The VM helper will give an assert if it's not (and has a comment about that in the prototype). I could put a similar comment for `isStable()`.

Reply:
Okay I'll let you decide if you want to add in the comment or not but given that there is no issue with a class not being initialized in the scenario that was being discussed above, I will approve this commit now.